zebra-chain/src/value_balance.rs

```diff
@@ -0,0 +1,297 @@
+//! A type that can hold the four types of Zcash value pools.
+
+use crate::amount::{Amount, Constraint, Error, NegativeAllowed, NonNegative};
+
+/// An amount spread between different Zcash pools.
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(bound = "C: Constraint")]
+pub struct ValueBalance<C = NegativeAllowed> {
```

Comment:
Better Reviews: Can we remove the default constraint here?
```suggestion
pub struct ValueBalance<C> {
```

There isn't really a sensible default constraint: shielded data value balances can be negative, but chain value balances must be positive.